### PR TITLE
fixup and enable condition checking tests

### PR DESCRIPTION
**Pull request by arvidn**
for the edge cases of absolute timestamp and absolute height conditions in mempool_manager

### CODE REVIEW

File: tests/core/full_node/test_mempool.py [hunk 1]
```diff
@@ -277,17 +280,17 @@ async def test_basic_mempool_manager(self, bt, two_nodes_mempool, wallet_a, self
...
-            # tests use the same ones. See comment at the two_nodes_mempool fixture
-            # (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 2, MempoolInclusionStatus.SUCCESS),
-            # (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 3, MempoolInclusionStatus.SUCCESS),
-            # (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 4, MempoolInclusionStatus.PENDING),
-            # (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 5, MempoolInclusionStatus.PENDING),
+            # run the test on. The fixture (two_nodes_mempool) creates 3 blocks,
+            # then condition_tester2 creates another 3 blocks
+            (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 4, MempoolInclusionStatus.SUCCESS),
+            (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 5, MempoolInclusionStatus.SUCCESS),
+            (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 6, MempoolInclusionStatus.PENDING),
```
Comment (aqk) [inline tests/core/full_node/test_mempool.py:287]: @mariano54  Is `PENDING` the correct return for `ASSERT_HEIGHT_ABSOLUTE 6` when chain height is 6?

Comment (arvidn) [reply tests/core/full_node/test_mempool.py:287]: note that this is testing the mempool. When inserting a spend bundle in the mempool, conditions that *will hold* in the future are still saved, in a separate *pending* queue. That's what the `PENDING` return means.
Testing the consensus (which is done in a different test) these same tests are already enabled.

**Comment by arvidn** [comment]
The reason I have to pass in `test=True` in so many places is because the full node uses wall clock time. I think that's problematic and we should move in a direction of abstracting that away instead. In such case, the clock node clock could be simulated just like the chain timestamps are.

https://github.com/Chia-Network/chia-blockchain/blob/main/chia/full_node/full_node.py#L652

File: tests/core/full_node/test_mempool.py (hunk 1 above)
Comment (aqk) [reply tests/core/full_node/test_mempool.py:289]: In a funny coincidence, I just made a change that involved `MempoolInclusionStatus`, so I now understand this section better.

File: chia/full_node/full_node_api.py [hunk 2]
```diff
@@ -1256,11 +1256,10 @@ async def request_removals(self, request: wallet_protocol.RequestRemovals) -> Op
         return msg
 
     @api_request
-    async def send_transaction(self, request: wallet_protocol.SendTransaction) -> Optional[Message]:
+    async def send_transaction(self, request: wallet_protocol.SendTransaction, test=False) -> Optional[Message]:
```
Comment (aqk) [inline chia/full_node/full_node_api.py:1259]:
```suggestion
    async def send_transaction(self, request: wallet_protocol.SendTransaction, *, test=False) -> Optional[Message]:
```

Comment (aqk) [reply chia/full_node/full_node_api.py:1259]: It's nice to force these named arguments to be named, especially boolean arguments.

File: tests/core/full_node/test_mempool.py [hunk 3]
```diff
@@ -277,17 +282,17 @@ async def test_basic_mempool_manager(self, bt, two_nodes_mempool, wallet_a, self
...
+            (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 7, MempoolInclusionStatus.PENDING),
             # genesis timestamp is 10000 and each block is 10 seconds
-            # (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, 10029, MempoolInclusionStatus.SUCCESS),
-            # (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, 10030, MempoolInclusionStatus.SUCCESS),
-            # (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, 10031, MempoolInclusionStatus.FAILED),
-            # (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, 10032, MempoolInclusionStatus.FAILED),
+            (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, 10049, MempoolInclusionStatus.SUCCESS),
```
Comment (aqk) [inline tests/core/full_node/test_mempool.py:292]: It's such a big jump from 7 to 10049 - why is the calculation so different from the 10029 it replaces?
(I understand why the low numbers changed).

Oh, right, these are seconds, and we are using 10-second intervals to quantize these steps. OK.

Sometimes it's nice to build up the tested value as an explanation (eg. `10000 + 49`), but I don't think that's needed here.

**Review by aqk — DISMISSED** [review]

File: tests/core/full_node/test_mempool.py (hunk 3 above)
Comment (arvidn) [reply tests/core/full_node/test_mempool.py:292]: the low numbers (4,5,6 and 7) are block *heights*. The large numbers (starting at 10000) are *seconds*. note that those tests are for different conditions

**Review by aqk — APPROVED** [review]
